conf/gulp-tasks/library.gulpfile.js

```diff
@@ -19,6 +20,23 @@ const uglify = require('gulp-uglify-es').default;
 
 const NAMESPACE = 'ANSWERS';
 
+function getLibVersion () {
+  const hash = require('child_process')
+    .execSync('git rev-parse HEAD')
+    .toString().trim();
+
+  const version = require('child_process')
+    .execSync('git describe --match "v[0-9]*" --abbrev=0 HEAD --always')
+    .toString().trim();
+
+  let libVersion = `canary-${hash}`;
```

Comment:
So is the idea here that any dev branches will have the libVersion set to canary-${commit sha}?

Reply:
I'm wondering if we should hold off on shipping this PR until we figure out what canary will look like in the CDN. 

@bryanreed have you and Christian discussed some ideas about canary and how it will appear in the CDN? I don't think we'd want to have assets in the CDN for each commit in the develop/canary branch. That would mean a pretty big increase in the number of assets we store in S3. I was envisioning we'd have a single canary target in the CDN that would get updated every time a new commit was merged into develop/canary.

Reply:
Hiya, as discussed, canary branches will only be added to the develop branch, but we will keep the hash so users can pin to them.